### ISSUE TITLE
Usability improvements for `tidy3d configure`

### ISSUE THREAD

**Issue by tylerflex** [flexcompute/tidy3d#657]
## Using password

Lei Z brought up a good point, which is that instead of 
```
tidy3d configure
```
prompting for the API key directly, it might be easier to have the user input their password at this stage, and then automatically generate and store the API for them.  This would avoid the user needing to log into the web interface to copy the API key.

If the user wanted to use the API key directly, they could add a flag, eg.
```
tidy3d configure --apikey
```

## Displaying previous API key

From Xin:

If there exists an API key already, the prompt gets quite messy as it is included in the prompt `API key[{previous_key}]`.
She suggests a prompt like

```
API Key:
Current: xxxxx
New:
```

**Comment by dbochkov-flexcompute** [comment]
a couple of comments:
- when an invalid apikey is entered during `tidy3d configure` config file is still created. Is this an intended behavior?
- when I have a correct `auth.json` and an invalid `config` then I get something strange. I wonder if something similar occurs if credentials are stored in environment variable
![Screenshot from 2023-01-10 16-56-28](https://user-images.githubusercontent.com/109110435/211681402-5b9fcb9d-6a87-4947-ab6f-5a0cfe1038ec.png)

**Comment by An-Li-magicloud** [comment]
``--apikey`` is supported already.
``` python
@click.command()
@click.option(
    "--apikey", prompt=config_description if "config_description" in globals() else "API Key"
)
def configure(apikey):
```

> a couple of comments:
> 
> * when an invalid apikey is entered during `tidy3d configure` config file is still created. Is this an intended behavior?
> * when I have a correct `auth.json` and an invalid `config` then I get something strange. I wonder if something similar occurs if credentials are stored in environment variable
>   ![Screenshot from 2023-01-10 16-56-28](https://user-images.githubusercontent.com/109110435/211681402-5b9fcb9d-6a87-4947-ab6f-5a0cfe1038ec.png)

Adjusted.

**Comment by tylerflex** [comment]
Fixed by #646
